python/ray/data/dataset.py

```diff
@@ -372,9 +375,9 @@ def map_batches(
             fn: The function to apply to each record batch, or a class type
                 that can be instantiated to create such a callable. Callable classes are
                 only supported for the actor compute strategy.
-            batch_size: The number of rows in each batch, or None to use entire blocks
-                as batches (blocks may contain different number of rows).
-                The final batch may include fewer than ``batch_size`` rows.
+            batch_size: The desired number of rows in each batch, or None to use entire
+                blocks as batches (blocks may contain different number of rows).
+                The actual batch size provided to fn may be smaller than ``batch_size``.
```

Comment:
This needs a bit explanation to set the expectation, user may ask "when" it'll be smaller etc.

Reply:
Yeah I tried finding a simple one-liner for "when", but it involves a lot of implementation details (e.g. blocks, the number of blocks per task), so I opted to tweak the existing line about when the batch can be smaller than ``batch_size``.

Got any ideas for concisely stating these semantics?

Reply:
I think we may leverage "map task" concept (already known by users), e.g. this should be true both before and after this change: if the number of records in the block (or blocks, if we do this PR) for a map task is less than batch_size, the actual batch provided to fn will be smaller than batch_size.

Reply:
I was a bit concerned about saying something like that since the user will wonder how we decide how many blocks are sent to a given map task, which blocks are sent to a given map task, etc. This also doesn't cover the case in which the number of records in the block(s) for a map task is **greater** than `batch_size`, but `num_records_in_blocks % batch_size != 0`, e.g. if `num_records_in_blocks = 11` and `batch_size = 10`, `fn` will get a 10 record batch followed by a 1 record batch.

Reply:
Added details to the docs for this arg + a note further up in the docstring.